File: DSA/HR/pangrams.cpp
#include <string>
#include <unordered_set>

using namespace std;

string pangrams(string s)
{
    std::unordered_set<char> seen{};

    for (char c : s)
    {
        if (c >= 'a' && c <= 'z' || c >= 'A' && c <= 'Z')
            seen.insert(tolower(c));
    }

    return seen.size() == 26 ? "pangram" : "not pangram";
}

File: DSA/HR/pangrams.java
import java.util.HashSet;

class Result {
    public static String pangrams(String s) {
        HashSet<Character> uniqueChars = new HashSet<>();

        for (char c : s.toCharArray()) {
            if (c != ' ') {
                uniqueChars.add(Character.toLowerCase(c));
            }
        }

        return uniqueChars.size() == 26 ? "pangram" : "not pangram";
    }

}